cider/exec/template/InValuesIR.cpp

```diff
@@ -226,3 +282,71 @@ std::unique_ptr<InValuesBitmap> CodeGenerator::createInValuesBitmap(
   }
   return nullptr;
 }
+
+// string handling is different compare to createInValuesBitmap
+std::unique_ptr<InValuesBitmap> CodeGenerator::createInValuesBitmapArrow(
+    const Analyzer::InValues* in_values,
+    const CompilationOptions& co) {
+  AUTOMATIC_IR_METADATA(cgen_state_);
+  const auto& value_list = in_values->get_value_list();
+  const auto val_count = value_list.size();
+  const auto& ti = in_values->get_arg()->get_type_info();
+  if (!(ti.is_integer() || (ti.is_string() && ti.get_compression() == kENCODING_DICT))) {
+    return nullptr;
+  }
+  const auto stringHasher =
+      ti.is_string() ? executor()->getCiderStringHasherHandle() : nullptr;
+  // FIXME: (yma11) in heavydb, use bitmap only when values count > 3
+  // and switch to normal OR op if not. We make this change because of
+  // type info check will fail for case substring(col, const_1, const_2) = "value"
+  // as substring has SQLTypeInfo(kTEXT, kENCODING_DICT, 0, kNULLT) while
+  // const_1/const_2 has SQLTypeInfo(kVARCHAR, kENCODING_NONE, 0, kNULLT)
+  if (val_count > 3 || ti.is_string()) {
```

Comment:
Since you have provided `codegenStringEq`, will `substring(col, const_1, const_2) = 'abc'` be supported? if so, we can revert the logic to remove `FIXME` here.

Reply:
such case is supported now, see [here](https://github.com/intel/BDTK/pull/144/files#diff-02bb724432fbc1bf10058038ebddc0fb5341065824c220fcdb896b6f3cd8a2aeR350) Let me update comments here.